src/app/import/components/upload/upload.component.ts

```diff
@@ -300,4 +303,13 @@ export class UploadComponent implements OnInit, OnChanges, OnDestroy {
     this.uploadSpecificHelper.selectedTokenizer = findName.name;
   }
 
+  prepareEmbeddings(): void {
+    let q, vc;
+    [q, vc] = this.projectApolloService.getEmbeddingSchema(this.projectId);
+    vc.pipe(first()).subscribe((embeddings: Embedding[]) => {
+      const hasNotGdprEmbeddings = embeddings.filter((e: Embedding) => e.platform == PlatformType.COHERE || e.platform == PlatformType.OPEN_AI).length > 0;
```

Comment:
since we don't actually need the whole list (just want to know if at least one exists) i'd go with `some` instead of a full filter (which in turn creates a new list etc.)